### PR TITLE
Add option for paren after func def

### PR DESCRIPTION
**Pull request by brmqk3**
The new option adds an indent to the open paren after a
function definition, if the paren is on a different line than 
the definition.

### CODE REVIEW

File: tests/input/cpp/indent_paren_after_func_def.cpp [hunk 1]
```diff
@@ -0,0 +1,5 @@
+void SomeFunction
+(
+int const aTest,
+int const aResult
+);
```
Comment (CDanU) [inline tests/input/cpp/indent_paren_after_func_def.cpp]: That is an example for a function declaration  ....

File: scripts/More_Options_to_Test/show_config.txt [hunk 2]
```diff
@@ -829,6 +829,9 @@ indent_paren_close              Unsigned Number
   1: Align under the open paren
   2: Indent to the brace level
 
+indent_paren_after_func_def     { False, True }
+  Controls the indent of the open paren after a function definition, if on it's own line.If True, indents the open paren
```
Comment (CDanU) [inline scripts/More_Options_to_Test/show_config.txt]: > __after__ a function definition

The parenthesis that are indented in your example are part of a function definition/declaration.

File: tests/input/cpp/indent_paren_after_func_def.cpp (hunk 1 above)
Comment (brmqk3) [inline tests/input/cpp/indent_paren_after_func_def.cpp]: It actually works the same for definition and declaration (due to how the code is already structured), but I understand what you're saying.

**Comment by CDanU** [comment]
You did not understand me correctly. The problem is not your test case but the faulty description of what it is your option is going to do. Do you want to support both function declarations and definitions? Or on of them? 

While you are at it could do the following: implement an (generic) option that works for both cases (I assume that is what currently your PR does), and add specialized options for each case that overwrite the generic option. 
See how some of the options that we already have, work only on func decls or func defs.

**Comment by brmqk3** [comment]
Alright, I added separate options for function calls, definitions, and declarations.

**Comment by CDanU** [comment]
Ok thanks, you kinda did one of the (optional) improvement suggestions. 
The main critique about this PR ,for now, still applies.

**Comment by brmqk3** [comment]
So the main critique is that the inputs aren't default formatted?

**Comment by CDanU** [comment]
And the faulty descriptions. Parenthesis are part of a function call/decl/def.

**Comment by brmqk3** [comment]
Ah, okay gotcha. Will fix that.

**Comment by CDanU** [comment]
Descriptions are OK now.
> Your tests are a bit short, what about function decl/def that are inside a a body(struct, class, namespace, ...)? The input file is not default formatted with an empty config.

__Still__ missing ... this kinda gets too tedious.

**Comment by brmqk3** [comment]
Sorry about that, forgot about the class. Alright, all of them are inside classes now.

**Pull request merged**